hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/HoodieSqlCommonUtils.scala

```diff
@@ -223,6 +223,18 @@ object HoodieSqlCommonUtils extends SparkAdapterSupport {
   def isHoodieConfigKey(key: String): Boolean =
     key.startsWith("hoodie.") || key == DataSourceReadOptions.TIME_TRAVEL_AS_OF_INSTANT.key
 
+  /**
+   * Extract hoodie config from conf using prefix "spark.hoodie." and "hoodie.".
+   * Priority for the same key: hoodie. > spark.hoodie.
+   */
+  def extractHoodieConfig(conf: Map[String, String]): Map[String, String] = {
+    val (withSparkPrefix, withoutSparkPrefix) = conf.partition(_._1.startsWith("spark."))
+    val combinedConf = withSparkPrefix.map {
+      case (key, value) => (key.stripPrefix("spark."), value)
```

Comment:
Should we handle case insensitive of key and value here?

Reply:
As I noticed the case insensitive check is more likely to be used in identifier compare like database name, table name etc. So I think it's OK to use original config here.